linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/java/org/apache/linkis/manager/am/util/LinkisUtils.java

```diff
@@ -0,0 +1,240 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.manager.am.util;
+
+import org.apache.linkis.common.exception.ErrorException;
+import org.apache.linkis.common.exception.FatalException;
+import org.apache.linkis.common.exception.WarnException;
+
+import java.time.Duration;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LinkisUtils {
+  private static final Logger logger = LoggerFactory.getLogger(LinkisUtils.class);
+
+  public static final ScheduledThreadPoolExecutor defaultScheduler =
+      new ScheduledThreadPoolExecutor(20, threadFactory("Linkis-Default-Scheduler-Thread-", true));
+
+  {
+    defaultScheduler.setMaximumPoolSize(20);
+    defaultScheduler.setKeepAliveTime(5, TimeUnit.MINUTES);
+  }
+
+  public static <T> T tryCatch(Callable<T> tryOp, Function<Throwable, T> catchOp) {
+    T call = null;
+    try {
+      call = tryOp.call();
+    } catch (Throwable t) {
+      if (t instanceof FatalException) {
+        logger.error("Fatal error, system exit...", t);
+        throw (FatalException) t;
+      } else if (t instanceof Error) {
+        logger.error("Throw error", t);
+        throw (Error) t;
+      } else {
+        call = catchOp.apply(t);
+      }
+    }
+    return call;
+  }
+
+  public static void tryFinally(Runnable tryOp, Runnable finallyOp) {
+    try {
+      tryOp.run();
+    } finally {
+      finallyOp.run();
+    }
+  }
+
+  public static <T> T tryAndWarn(Callable<T> tryOp, Logger log) {
+    return tryCatch(
+        tryOp,
+        t -> {
+          if (t instanceof ErrorException) {
+            ErrorException error = (ErrorException) t;
+            log.warn(
+                "Warning code（警告码）: {}, Warning message（警告信息）: {}.",
+                error.getErrCode(),
+                error.getDesc(),
+                error);
+
+          } else if (t instanceof WarnException) {
+            WarnException warn = (WarnException) t;
+            log.warn(
+                "Warning code（警告码）: {}, Warning message（警告信息）: {}.",
+                warn.getErrCode(),
+                warn.getDesc(),
+                warn);
+
+          } else {
+            log.warn("", t);
+          }
+          return null;
+        });
+  }
+
+  public static void tryAndErrorMsg(Runnable tryOp, String message, Logger log) {
+    try {
+      tryOp.run();
+    } catch (WarnException t) {
+      WarnException warn = (WarnException) t;
+      log.warn(
+          "Warning code（警告码）: {}, Warning message（警告信息）: {}.", warn.getErrCode(), warn.getDesc());
```

Comment:
Is this method the corresponding org.apache.linkis.common.utils.Utils#tryAndErrorMsg?

Reply:
Yes, this method correspond to org.apache.linkis.common.utils.Utils#tryAndErrorMsg.